website/www/site/content/en/documentation/patterns/bqml.md

```diff
@@ -0,0 +1,98 @@
+OPTIONS(model_type='logistic_reg', input_label_cols=["label"]) AS
+SELECT
+  IF(totals.transactions IS NULL, 0, 1) AS label,
+  IFNULL(geoNetwork.country, "") AS country
+FROM
+  `bigquery-public-data.google_analytics_sample.ga_sessions_*`
+WHERE
+  _TABLE_SUFFIX BETWEEN '20160801' AND '20170630'
+```
+
+The model will predict if a purchase will be made given the country of the visitor on data gathered between 2016-08-01 and 2017-06-30.
+
+## Export your BigQuery ML model
+
+In order to incorporate your model in an Apache Beam pipeline, you will need to export it. Prerequisites to do so are [installing the `bq` command-line tool](https://cloud.google.com/bigquery/docs/bq-command-line-tool) and [creating a Google Cloud Storage bucket](https://cloud.google.com/storage/docs/creating-buckets) to store your exported model.
+
+Export the model using the following command:
+
+```bash
+bq extract -m bqml_tutorial.sample_model gs://some/gcs/path
+```
+
+## Create an Apache Beam transform that uses your BigQuery ML model
+
+In this section we will construct an Apache Beam pipeline that will use the BigQuery ML model we just created and exported. The model can be served using Google Cloud AI Platform Prediction - for this please refer to the [AI Platform patterns](https://beam.apache.org/documentation/patterns/ai-platform/). In this case, we'll be illustrating how to use the tfx_bsl library to do local predictions (on your Apache Beam workers).
+
+First, the model needs to be downloaded to a local directory where you will be developing the rest of your pipeline (e.g. to `serving_dir/sample_model/1`).
+
+Then, you can start developing your pipeline like you would normally do. We will be using the `RunInference` PTransform from the [tfx_bsl](https://github.com/tensorflow/tfx-bsl) library, and we will point it to our local directory where the model is stored (see the `model_path` variable in the code example). The transform takes elements of the type `tf.train.Example` as inputs and outputs elements of the type `tensorflow_serving.apis.prediction_log_pb2.PredictionLog`.
+
+```python
+import apache_beam
+import tensorflow as tf
+from google.protobuf import text_format
+from tfx_bsl.beam import run_inference
+from tfx_bsl.public.beam import RunInference
+from tfx_bsl.public.proto import model_spec_pb2
+
+
+inputs = """
+features {
+    feature { key: "country" value { bytes_list { value: 'Belgium' }}}
+}
+"""
+
+def create_tf_example(json_obj):
+    return text_format.Parse(json_obj, tf.train.Example())
+
+model_path = "serving_dir/sample_model/1"
+
+with beam.Pipeline() as p:
+    res = (
+        p
+        | beam.Create([
+            create_tf_example(inputs)
+        ])
+        | RunInference(
+            model_spec_pb2.InferenceSpecType(
+                saved_model_spec=model_spec_pb2.SavedModelSpec(
+                    model_path=model_path,
+                    signature_name=['serving_default'])))
+```
```

Comment:
It would be useful to show how to parse the results coming from RunInference which is a predict log.